package-build.el

```diff
@@ -91,10 +93,17 @@
   :group 'package-build
   :type 'boolean)
 
+(defcustom package-build-elpa-versioning-scheme nil
+  "When non-nil, use GNU ELPA Devel versioning scheme for non-stable builds."
+  :group 'package-build
+  :type 'boolean)
+
 (defcustom package-build-get-version-function
   (if package-build-stable
       'package-build-get-tag-version
-    'package-build-get-timestamp-version)
+    (if package-build-elpa-versioning-scheme
+        'package-build-get-tag-and-timestamp-version
+      'package-build-get-timestamp-version))
```

Comment:
No new option is needed. Just set `package-build-get-version-function` instead. It was created with the intention to allow the use of arbitrary function extraction functions without having to add a new option for each one of them.